pkg/services/authn/authnimpl/sync/rbac_sync_test.go

```diff
@@ -421,6 +422,129 @@ func TestRBACSync_ClearUserPermissionCacheHook(t *testing.T) {
+			k8sPerms: []string{
+				"folder.grafana.app/folders:get",
+			},
+			expectedPerms: []accesscontrol.Permission{
+				{Action: "folders:read", Scope: "folders:uid:*"},
+			},
+			expectedError: nil,
+		},
+		{
+			name: "should translate resource with wildcard verb",
+			k8sPerms: []string{
+				"folder.grafana.app/folders:*",
+			},
+			expectedPerms: []accesscontrol.Permission{
+				{Action: "folders:read", Scope: "folders:uid:*"},
+				{Action: "folders:write", Scope: "folders:uid:*"},
+				{Action: "folders:delete", Scope: "folders:uid:*"},
+				{Action: "folders:create", Scope: "folders:uid:*"},
+				{Action: "folders.permissions:read", Scope: "folders:uid:*"},
+				{Action: "folders.permissions:write", Scope: "folders:uid:*"},
+			},
+			expectedError: nil,
+		},
+		{
+			name: "should handle invalid permission format",
+			k8sPerms: []string{
+				"invalid-format",
+			},
+			expectedPerms: []accesscontrol.Permission{},
+			expectedError: nil,
+		},
+		{
+			name: "should handle unknown resource",
+			k8sPerms: []string{
+				"folder.grafana.app/unknown:get",
+			},
+			expectedPerms: []accesscontrol.Permission{},
+			expectedError: nil,
+		},
+		{
+			name: "should handle unknown verb",
+			k8sPerms: []string{
+				"folder.grafana.app/folders:unknown",
+			},
+			expectedPerms: []accesscontrol.Permission{},
+			expectedError: nil,
+		},
+		{
+			name: "should handle group:verb format",
+			k8sPerms: []string{
+				"folder.grafana.app:get",
+			},
+			expectedPerms: []accesscontrol.Permission{
+				{Action: "folders:read", Scope: "folders:uid:*"},
+			},
+			expectedError: nil,
+		},
+		{
+			name: "should handle group:* format",
+			k8sPerms: []string{
+				"folder.grafana.app:*",
+			},
+			expectedPerms: []accesscontrol.Permission{
+				{Action: "folders:read", Scope: "folders:uid:*"},
+				{Action: "folders:write", Scope: "folders:uid:*"},
+				{Action: "folders:delete", Scope: "folders:uid:*"},
+				{Action: "folders:create", Scope: "folders:uid:*"},
+				{Action: "folders.permissions:read", Scope: "folders:uid:*"},
+				{Action: "folders.permissions:write", Scope: "folders:uid:*"},
+			},
+			expectedError: nil,
+		},
+		{
+			name: "should handle unknown group in group:verb format",
+			k8sPerms: []string{
+				"unknown.grafana.app:get",
+			},
+			expectedPerms: []accesscontrol.Permission{},
+			expectedError: nil,
+		},
+		{
+			name: "should handle unknown verb in group:verb format",
+			k8sPerms: []string{
+				"folder.grafana.app:unknownverb",
+			},
+			expectedPerms: []accesscontrol.Permission{},
+			expectedError: nil,
+		},
+		{
+			name: "should handle multiple group:verb permissions",
+			k8sPerms: []string{
+				"folder.grafana.app:get",
+				"folder.grafana.app:create",
+			},
+			expectedPerms: []accesscontrol.Permission{
+				{Action: "folders:read", Scope: "folders:uid:*"},
+				{Action: "folders:create", Scope: "folders:uid:*"},
+			},
+			expectedError: nil,
+		},
```

Comment:
Great testing 💯
I suggest we also cover the invalid cases are skipped :)
```suggestion
		},
		{
			name: "should skip invalid permissions group/resource/additional/part:verb",
			k8sPerms: []string{
				"folder.grafana.app/folder/fold1/subresource:get",
			},
			expectedPerms: []accesscontrol.Permission{},
		},
```